src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -826,6 +836,42 @@ fn rustflags_from_target(
     }
 }
 
+/// Gets the user-specified linker for a particular host or target from the configuration.
+fn target_linker(
```

Comment:
Good catch! I entirely missed that this was a problem.

I don't see a good solution to this, I'll come back with a fresh set of eyes tomorrow and see if that changes.

Right now my take is it's either this (bad), don't fix this issue at all (also bad), or doing something involving not so clean code to defer actually returning the error (also bad). But I think there's a small chance I can find a better way.

Reply:
Honestly, the current approach is not too bad for normal users.

However, I know some enterprise users have weird multi-level configuration setup. And may move projects among different directories. This is also bad because you cannot opt out directory-probing for configuration files (which Cargo may provide an escape patch?).